### PR TITLE
feature(api-server): feature wasi builder

### PR DESCRIPTION
**Pull request by oojimmy**
## Issue
- [x] https://github.com/tricorder-observability/Starship/issues/187

### CODE REVIEW

File: src/api-server/wasm/wasm.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/tricorder/src/utils/errors"
+	"github.com/tricorder/src/utils/uuid"
+)
+
+const (
+	WASIClang           = "WASI_SDK_PATH=/opt/wasi-sdk /opt/wasi-sdk/bin/clang"
```
Comment (nascentcore-eng) [inline src/api-server/wasm/wasm.go]: We probably should package wasi-sdk under `/starship`, i.e.:
`/opt/wasi-sdk` changed to `/starship/wasm/wasi-sdk`

Like /starshp/linux-headers, we can put all things packaged by starship under /starship.

File: src/api-server/wasm/wasm.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	WASICFlags          = "--sysroot=/opt/wasi-sdk/share/wasi-sysroot -Wall -Wextra -Wl,--export-all"
```
Comment (nascentcore-eng) [inline src/api-server/wasm/wasm.go]: `/opt/wasi-sdk` changed to `/starship/wasm/wasi-sdk`

File: src/api-server/wasm/wasm.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	WASIStarshipInclude = "/opt/starship/include"
```
Comment (nascentcore-eng) [inline src/api-server/wasm/wasm.go]: Should `/opt/starship` changed to `/starship/wasm/include`
I.e., this is an include for wasm, for /starship

File: src/api-server/wasm/wasm.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+	// Read the first four bytes of the file
+	buf := make([]byte, 4)
+	n, err := file.Read(buf)
+	if err != nil || n < 4 {
+		return false
+	}
+
+	// Compare with wasm magic number: \x00\x61\x73\x6d (0x6d736100 in little endian)
+	wasmMagic := []byte{0x00, 0x61, 0x73, 0x6d}
```
Comment (nascentcore-eng) [inline src/api-server/wasm/wasm.go]: Might move `isWasmFile()` into `src/utils/wasm/file.go`

File: src/api-server/wasm/wasm.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+		if buf[i] != wasmMagic[i] {
+			isWasm = false
+			break
+		}
+	}
+
+	return isWasm
+}
+
+func WASICompileC(code string) ([]byte, error) {
```
Comment (nascentcore-eng) [inline src/api-server/wasm/wasm.go]: As we suggested, define type WASICompiler, and add WasiCompiler.BuildC().

**Comment by oojimmy** [comment]
waiting for https://github.com/tricorder-observability/Starship/pull/218 was merged

**Review by nascentcore-eng — APPROVED** [review]

**Pull request merged**